packages/eslint-plugin/configs/recommended.js

```diff
@@ -13,8 +13,17 @@ const defaultPrettierConfig = require( '@wordpress/prettier-config' );
  */
 const { isPackageInstalled } = require( '../utils' );
 
-const { config: localPrettierConfig } =
+let { config: localPrettierConfig } =
 	cosmiconfigSync( 'prettier' ).search() || {};
```

Comment:
In the `cosmiconfig` docs they always pass the name of the module to `cosmiconfigSync`. Here it's omitted when calling load. I don't know if it matters but I thought I will raise it. I was wondering if we could consolidate the logic with their promise-based API for `search`, but I'm not clear if the following is more readable. 


```js
const explorerSync = cosmiconfigSync( 'prettier' );

const localPrettierConfig = explorerSync.search().then( ( config, filepath, isEmpty ) => {
    if ( ! isEmpty ) {
        return config;   
    }
    return explorerSync.load( filepath ) || {};
} );
```

Reply:
I studied documentation a bit more and it seems like a bug that `config` would keep the path to the file 🤷🏻

Reply:
> Here it's omitted when calling load. I don't know if it matters but I thought I will raise it.

It doesn't _really_ matter since `load` doesn't need to know anything to just load the file we pass, but for consistency with the docs I can add it back.

Your code doesn't work exactly, because cosmiconfig considers this config value to be valid - so `isEmpty` is undefined, `filePath` is where the config is found, `/path/to/package.json`, and `config` itself is `./.prettierrc.js`.

Another idea is to use the `transform` option, like this:

```js
const explorerSync = cosmiconfigSync( 'prettier', {
	transform: function( result ) {
		const { config } = result;
		if ( typeof config === 'string' ) {
			return (
				cosmiconfigSync( 'prettier' ).load( config ) || {}
			)
		}
		return result;
	}
} );
const { config: localPrettierConfig } = explorerSync.search()
```

This works with my test repo… but I've also tested with a plain shared config (eg `"prettier": "@company/prettier-config"`) - that has the same issue (config imported as a string) but since that isn't a local file, `load` doesn't find it.

> I studied documentation a bit more and it seems like a bug that config would keep the path to the file 

It does seem strange… I can open an issue in their repo about it, and see if it is a bug.

Reply:
Thank you for exploring it further. I'm fine with the current proposal with a follow-up report in the upstream repository.